src/Cli/dotnet/commands/dotnet-workload/install/NetSdkMsiInstallerClient.cs

```diff
@@ -376,15 +428,42 @@ public WorkloadSet InstallWorkloadSet(ITransactionContext context, string worklo
             }
         }
 
-        private void RemoveWorkloadManifests(List<WorkloadManifestRecord> manifestToRemove, DirectoryPath? offlineCach)
+        private void RemoveWorkloadSets(List<WorkloadSetRecord> workloadSetsToRemove, DirectoryPath? offlineCache)
+        {
+            foreach (WorkloadSetRecord record in workloadSetsToRemove)
+            {
+                DetectState state = DetectPackage(record.ProductCode, out Version _);
+                if (state == DetectState.Present)
+                {
+                    string msiNuGetPackageId = $"Microsoft.NET.Workloads.{record.WorkloadSetFeatureBand}.Msi.{HostArchitecture}";
+                    string packageVersion = WorkloadSet.WorkloadSetVersionToWorkloadSetPackageVersion(record.WorkloadSetVersion, out _);
+                    MsiPayload msi = GetCachedMsiPayload(msiNuGetPackageId, packageVersion, offlineCache);
+
+                    if (!string.Equals(record.ProductCode, msi.ProductCode, StringComparison.OrdinalIgnoreCase))
+                    {
```

Comment:
This is a case where we have two different set MSIs describing the set - we're just logging this as an FYI for the user?

Reply:
Yeah, I don't think we expect this to happen, but I was just copying the pattern for other MSI types that we uninstall.